### PR TITLE
Support for building arm64 MSIs

### PR DESCRIPTION
**Pull request by NikolaMilosavljevic**
This change adds arm64 Wix support for Shared FX in Arcade, it allows other repos to build arm64 MSIs and Burn bundles. No additional changes are needed for dotnet/runtime and dotnet/windowsdesktop repos, besides picking up a new version of Arcade, that includes this change.

This fixes https://github.com/dotnet/runtime/issues/1529 and expands further on the original plan. We will produce native arm64 MSI, and EXE, packages.

Arm64 requires v5.0 of Windows Installer schema - https://docs.microsoft.com/en-us/windows/win32/msi/using-64-bit-windows-installer-packages

Schema for other architectures is unchanged (v2.0) to allow installation on the same supported set of OS targets.

### CODE REVIEW

**Comment by hoyosjs** [comment]
cc: @tommcdon

File: src/Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk/targets/windows/wix.targets [hunk 1]
```diff
@@ -183,10 +183,17 @@
             GetBundleWixConfiguration">
     <PropertyGroup>
       <OutInstallerFile Condition="'$(OutInstallerFile)' == ''">$(InstallerFile)</OutInstallerFile>
+      <InstallerVersion>200</InstallerVersion>
+      <!--
+        Native arm64 MSI packages require version 500 of MSI database schema, see:
+        https://docs.microsoft.com/en-us/windows/win32/msi/using-64-bit-windows-installer-packages
+      -->
+      <InstallerVersion Condition="'$(MsiArch)' == 'arm64'">500</InstallerVersion>
```
Comment (dagood) [inline src/Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk/targets/windows/wix.targets:191]: I'd leave this open to override (so the theoretical user doesn't need to do something extreme like `Remove` the item and add a new one in a target):

```xml
      <!--
        Native arm64 MSI packages require version 500 of MSI database schema, see:
        https://docs.microsoft.com/en-us/windows/win32/msi/using-64-bit-windows-installer-packages
      -->
      <InstallerVersion Condition="'$(InstallerVersion)' == '' and '$(MsiArch)' == 'arm64'">500</InstallerVersion>
      <InstallerVersion Condition="'$(InstallerVersion)' == ''">200</InstallerVersion>
```

Comment (NikolaMilosavljevic) [reply src/Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk/targets/windows/wix.targets]: I've pushed the change - thanks Davis!

**Comment by jkoritzinsky** [comment]
I'll need to make sure to port this over to the new Shared Framework SDK that I'm working on.

**Comment by tommcdon** [comment]
ARM64 MSI testing complete and signing off.  Thank you Nikola!

**Pull request merged**